### PR TITLE
feat: DistributingExecutor fails DDL statement if CommandRunner DEGRADED

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
Stack on top of https://github.com/confluentinc/ksql/pull/6012
Changes the DistributingExecutor to take a Supplier<Boolean> so that it can check the status of the CommandRunner and not to produce to the command topic if the status is DEGRADED.

### Testing done 
Make sure unit tests all still pass after refactor
Tried issuing DDL with server in DEGRADE state
```
ksql> create stream test_query as select * from test;
Failed to handle Ksql Statement.
The server has encountered an incompatible entry in its log and cannot process further statements.
This is most likely due to the service being rolled back to an earlier version.
```

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java [hunk 1]
```diff
@@ -61,14 +62,15 @@
 
   public DistributingExecutor(
       final KsqlConfig ksqlConfig,
-      final CommandQueue commandQueue,
+      final CommandRunner commandRunner,
```
Comment (rodesai) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java:65]: can we abstract this out as a `Supplier<Boolean>` instead?

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/DefaultErrorMessages.java [hunk 2]
```diff
@@ -19,6 +19,15 @@
 
 public class DefaultErrorMessages implements ErrorMessages {
 
+  static String COMMAND_RUNNER_DEGRADED_ERROR_MESSAGE =
```
Comment (rodesai) [inline ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/DefaultErrorMessages.java:22]: I'd drop the references to `CommandRunner` or command topic here as they are implementation details of the server that we shouldn't leak in the error message. How about something like:
`Failed to process KSQL statement. The server has encountered an incompatible entry in its log and cannot process further statements. This is most likely due to the service being rolled back to an earlier version.`

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java (hunk 1 above)
Comment (stevenpyzhang) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java]: done

File: ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/DefaultErrorMessages.java (hunk 2 above)
Comment (stevenpyzhang) [reply ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/DefaultErrorMessages.java:22]: I've updated the error message. I'm going to reuse `COMMAND_RUNNER_DEGRADED_ERROR_MESSAGE` for adding warnings later on to all responses from the `KsqlResource` so I moved "Failed to process KSQL statement" to the `DistributingExecutor`.

**Pull request merged**